articles/architecture-scenarios/implementation/b2c/b2c-authorization.md

```diff
@@ -12,26 +12,84 @@ useCase:
+Access Control is essentially a super-set of the authorization process; authorization in it's fundemantal form can be thought of as a coarse-grained operation, whereas Access Control provides much more fine-grained control. Whatever your use case, there are a number of things you will want to consider when looking at functionality and workflow:
+
+* Are there scenarios where access to an application or API should be rejected?
+* Will I be providing APIs that can be accessed by third-party applications?
+* Will my APIs also be accessed by our own (first-party) applications?
+* Will my application be calling a third-party API?
+* Should my applications and/or APIs be enforcing role or permission based access control?
+ 
+Auth0 provides coarse-grained authorization by restricting access to certain applications or APIs based on certain conditions. In a coarse-grained authorization scenario, a [Rule](/rules) can be built that returns an `UnauthorizedError` when, for example, a user attempts access to an application or an API at an incorrect time (as described in this [example](/authorization/concepts/sample-use-cases-rules#allow-access-only-on-weekdays-for-a-specific-application)) - or if the user doesn’t have the right claim(s) contained in their [Metadata](/users/concepts/overview-user-metadata). For an _application_ using OpenID Connect ([OIDC](/protocols/oidc)), this would prevent the allocation of the [ID Token](/tokens/id-token) used to authorize access. Similarly, for an _API_, allocation of any OAuth2 [Access Tokens](/tokens/overview-access-tokens) (used when [calling the API](/api-auth/why-use-access-tokens-to-secure-apis)), could be prevented as described in this [example](/api-auth/restrict-access-api#example-deny-access-to-anyone-calling-the-api).
+
+::: panel Best Practice
+In the same way that [OIDC](/protocols/oidc) is the most commonly used industry-standard protocol for authentication in customer facing applications, we find that [OAuth2](protocols/oauth2) is the most commonly used industry-standard protocol for authorization.
+:::
+
+Auth0 also provides fine-grained access control at both an application level and an API level. For [application level integration](#application-integration), Auth0 allows you to add [custom claims](#id-token-claims) to an ID Token, which your application can then verifiy and subsequently use to enforce access control. In this case you will need to decide what information you require for your application to make access control decisions.
+
+::: warning
+When deciding what data to include in OIDC tokens, consider token size, especially if you are passing the token in the URL. Even if you are not passing tokens in the URL, you will also need to consider the potential of exposing sensitive PII (Personally Identifiable Information).
```

Comment:
```suggestion
When deciding what data to include in OIDC tokens, consider token size, especially if you are passing the token in the URL. Even if you are not passing tokens in the URL, you will also need to consider the potential of exposing sensitive PII (Personally Identifiable Information).  Token information is not encrypted, so, though it isn't generally a security issue for an ID token to be leaked, it can be a privacy issue depending on the data that is included in the token.
```